### PR TITLE
Az/logger fix

### PR DESCRIPTION
**Pull request by azhavoro**

### CODE REVIEW

File: cvat/apps/engine/static/engine/js/annotationUI.js [hunk 1]
```diff
@@ -49,24 +49,14 @@
 */
 
 async function initLogger(jobID) {
-    if (!Logger.initializeLogger('CVAT', jobID)) {
+    if (!Logger.initializeLogger(jobID)) {
         const message = 'Could not initialize Logger. Please immediately report the problem to support team';
```
Comment (bsekachev) [inline cvat/apps/engine/static/engine/js/annotationUI.js]: const message = 'Logger has been already initialized'

I don't see any others scenarios for that condition.

File: cvat/apps/engine/static/engine/js/logger.js [hunk 2]
```diff
@@ -55,131 +52,117 @@ class LogCollection extends Array {
...
+class LoggerHandler {
+    constructor(jobId) {
+        this._clientID = Date.now().toString().substr(-6);
+        this._jobId = jobId;
+        this._username = null;
+        this._userActivityHandler = null;
+        this._logEvents = [];
+        this._userActivityHandler = new UserActivityHandler();
+        this._timeThresholds = {};
+        this.isInitialized = Boolean(this._userActivityHandler);
```
Comment (bsekachev) [inline cvat/apps/engine/static/engine/js/logger.js]: Always true

File: cvat/apps/engine/static/engine/js/logger.js [hunk 3]
```diff
@@ -375,11 +395,10 @@ var Logger = {
...
      * @static
      */
-    initializeLogger: function(applicationName, taskId)
-    {
+    initializeLogger: function(jobId) {
         if (!this._logger)
         {
-            this._logger = new LoggerHandler(applicationName, taskId);
+            this._logger = new LoggerHandler(jobId);
             return this._logger.isInitialized;
```
Comment (bsekachev) [inline cvat/apps/engine/static/engine/js/logger.js]: Always true

File: cvat/apps/engine/static/engine/js/logger.js [hunk 4]
```diff
@@ -55,131 +52,117 @@ class LogCollection extends Array {
...
-
-    this.addEvent = function(event)
-    {
-        this._pushEvent(event);
-    };
+class LoggerHandler {
+    constructor(jobId) {
+        this._clientID = Date.now().toString().substr(-6);
+        this._jobId = jobId;
+        this._username = null;
```
Comment (nmanovic) [inline cvat/apps/engine/static/engine/js/logger.js]: Please remove the field.

File: cvat/apps/engine/static/engine/js/logger.js [hunk 5]
```diff
@@ -55,131 +52,117 @@ class LogCollection extends Array {
...
-                else {
+        return new Promise((resolve, reject) => {
+            let retries = 3;
+            let makeRequest = () => {
+                let xhr = new XMLHttpRequest();
+                xhr.open('POST', '/api/v1/server/exception');
+                xhr.setRequestHeader('Content-Type', 'application/json');
+                xhr.setRequestHeader("X-CSRFToken", Cookies.get('csrftoken'));
+                let onreject = () => {
+                    if (--retries) {
```
Comment (bsekachev) [inline cvat/apps/engine/static/engine/js/logger.js]: retries --
In other case we have actually "retries - 1" retrise

File: cvat/apps/engine/serializers.py [hunk 6]
```diff
@@ -345,7 +345,7 @@ class LogEventSerializer(serializers.Serializer):
 
     name = serializers.CharField(max_length=64)
     time = serializers.DateTimeField()
-    message = serializers.CharField(max_length=4096, required=False)
+    message = serializers.CharField(max_length=4096, required=False, allow_blank=True)
```
Comment (nmanovic) [inline cvat/apps/engine/serializers.py]: remove allow_blank=True. Just don't include the field.

File: cvat/apps/engine/static/engine/js/logger.js [hunk 7]
```diff
@@ -55,131 +52,117 @@ class LogCollection extends Array {
...
-    this.addEvent = function(event)
-    {
-        this._pushEvent(event);
-    };
+class LoggerHandler {
+    constructor(jobId) {
+        this._clientID = Date.now().toString().substr(-6);
+        this._jobId = jobId;
+        this._username = null;
+        this._userActivityHandler = null;
```
Comment (nmanovic) [inline cvat/apps/engine/static/engine/js/logger.js]: Below we initialize the same field.

**Pull request merged**